src/python/pants/backend/python/rules/hermetic_pex.py

```diff
@@ -1,64 +1,117 @@
-        description: str,
-        input_files: Digest,
-        env: Optional[Mapping[str, str]] = None,
-        **kwargs: Any
-    ) -> ExecuteProcessRequest:
-        """Creates an ExecuteProcessRequest that will run a PEX hermetically.
-
-        :param python_setup: The parameters for selecting python interpreters to use when invoking
-                             the PEX.
-        :param subprocess_encoding_environment: The locale settings to use for the PEX invocation.
-        :param pex_path: The path within `input_files` of the PEX file (or directory if a loose
-                         pex).
-        :param pex_args: The arguments to pass to the PEX executable.
-        :param description: A description of the process execution to be performed.
-        :param input_files: The files that contain the pex itself and any input files it needs to
-                            run against.
-        :param env: The environment to run the PEX in.
-        :param **kwargs: Any additional :class:`ExecuteProcessRequest` kwargs to pass through.
-        """
-
-        # NB: we use the hardcoded and generic bin name `python`, rather than something dynamic like
-        # `sys.executable`, to ensure that the interpreter may be discovered both locally and in remote
-        # execution (so long as `env` is populated with a `PATH` env var and `python` is discoverable
-        # somewhere on that PATH). This is only used to run the downloaded PEX tool; it is not
-        # necessarily the interpreter that PEX will use to execute the generated .pex file.
-        # TODO(#7735): Set --python-setup-interpreter-search-paths differently for the host and target
-        # platforms, when we introduce platforms in https://github.com/pantsbuild/pants/issues/7735.
-        argv = ("python", pex_path, *pex_args)
-
-        hermetic_env = dict(
-            PATH=create_path_env_var(python_setup.interpreter_search_paths),
-            PEX_ROOT="./pex_root",
-            PEX_INHERIT_PATH="false",
-            PEX_IGNORE_RCFILES="true",
-            **subprocess_encoding_environment.invocation_environment_dict
-        )
-        if env:
-            hermetic_env.update(env)
+@dataclass(frozen=True)
+class HermeticPexRequest:
+    pex_path: str
+    exe_req: ExecuteProcessRequest
+
+
+class HermeticPex(Subsystem):
+    options_scope = "hermetic-pex-creation"
 
-        return ExecuteProcessRequest(
-            argv=argv, input_files=input_files, description=description, env=hermetic_env, **kwargs
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--python-executable-name",
+            type=str,
+            default="python",
+            help="The python executable to search for in --python-setup-interpreter-search-paths "
+            "when creating a pex with the v2 python backend.",
         )
+        register(
+            "--use-pex-cache",
```

Comment:
Why is this now an option? I believe it should always be `False` for hermiticity. 

It's likely best to save this particular option for a followup PR so that it can get the proper discussion it deserves and this PR can focus on the big wins of Pex EPRs now being rules + `--python-executable-name` being configurable.

Reply:
I think it's pretty reasonable to make this into an option since it makes running pex faster in some cases, which matters to real users, although we should also be setting PEX_ROOT in that case. I'll avoid adding it here because that's not a discussion I want to deal with right now.

Reply:
+1. If you have a use for this, it's a cheat / breaks the engine sandbox model.